### PR TITLE
Migrate Segment integration to react to site_view events

### PR DESCRIPTION
**Pull request by spastorelli**
This PR aims to migrate the Segment integration to react to `site_view` integration events.

### CODE REVIEW

File: integrations/segment/gitbook-manifest.yaml [hunk 1]
```diff
@@ -12,6 +12,7 @@ organization: gitbook
 script: ./src/index.ts
 scopes:
     - space:views:read
```
Comment (SamyPesse) [inline integrations/segment/gitbook-manifest.yaml:14]: Can we remove this one?

File: integrations/segment/gitbook-manifest.yaml [hunk 2]
```diff
@@ -33,6 +34,14 @@ summary: |
...
     - analytics
 configurations:
+    site:
+        properties:
+            write_key:
+                type: string
+                title: Write Key
+                description: Key provided by Segment to identify the data source.
+        required:
+            - write_key
```
Comment (SamyPesse) [inline integrations/segment/gitbook-manifest.yaml:44]: Does it mean that users will have to reconfigure or can we run a migration?

**Review by SamyPesse — APPROVED** [review]
Looks good, but you need to run `bun run changeset` for this package

File: integrations/segment/gitbook-manifest.yaml (hunk 2 above)
Comment (spastorelli) [reply integrations/segment/gitbook-manifest.yaml:44]: yeah we will have to run a migration to migrate space level installation to their sites

**Pull request merged**